File: application/src/test/java/com/smtm/application/security/TestSecurityConfiguration.java
package com.smtm.application.security;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import com.smtm.application.security.v1.TokenController;
import com.smtm.application.security.v1.UsersController;
import com.smtm.security.api.Authentication;
import com.smtm.security.api.Authorization;
import com.smtm.security.api.UserRegistration;

@Configuration
public class TestSecurityConfiguration {

    @Bean
    public UserRegistration userRegistration() {
        return new FakeUserRegistration();
    }

    @Bean
    public Authorization authorization(@Value("${smtm.security.jwt.secret}") String secret) {
        return new FakeAuthorization(secret);
    }

    @Bean
    public Authentication authentication(@Value("${smtm.security.jwt.secret}") String secret) {
        return new FakeAuthentication(secret);
    }

    @Bean
    public UsersController usersController(UserRegistration userRegistration) {
        return new UsersController(userRegistration);
    }

    @Bean
    public TokenController tokenController(Authentication authentication) {
        return new TokenController(authentication);
    }
}

File: application/src/test/kotlin/com/smtm/application/security/TestSecurityConfiguration.kt
package com.smtm.application.security

import com.smtm.application.security.v1.TokensController
import com.smtm.application.security.v1.UsersController
import com.smtm.security.api.CredentialsAuthentication
import com.smtm.security.api.Authorization
import com.smtm.security.api.RefreshTokenAuthentication
import com.smtm.security.api.UserRegistration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class TestSecurityConfiguration {

    @Bean
    fun userRegistration(): UserRegistration = FakeUserRegistration()

    @Bean
    fun authorization(): Authorization = FakeAuthorization()

    @Bean
    fun credentialsAuthentication(): CredentialsAuthentication = FakeCredentialsAuthentication()

    @Bean
    fun refreshTokenAuthentication(): RefreshTokenAuthentication = FakeRefreshTokenAuthentication()

    @Bean
    fun usersController(userRegistration: UserRegistration): UsersController = UsersController(userRegistration)

    @Bean
    fun tokenController(credentialsAuthentication: CredentialsAuthentication,
                        refreshTokenAuthentication: RefreshTokenAuthentication): TokensController = TokensController(credentialsAuthentication, refreshTokenAuthentication)

    @Bean
    fun testProtectedResourceController(): TestProtectedResourceController = TestProtectedResourceController()
}
